quipucords/api/serializers.py

```diff
@@ -39,3 +39,46 @@ def validate(self, attrs):
             raise ValidationError(_('ssh_keyfile, %s, is not a valid file'
                                     ' on the system.' % (ssh_keyfile)))
         return attrs
+
+
+class NetworkProfileSerializer(ModelSerializer):
+    """Serializer for the NetworkProfile model"""
+    class Meta:
+        model = NetworkProfile
+        fields = '__all__'
+
+    @staticmethod
+    def validate_name(name):
+        """Validate the name of the NetworkProfile."""
+        if not isinstance(name, str) or not name.isprintable():
+            raise ValidationError('NetworkProfile must have printable name.')
+
+        return name
+
+    @staticmethod
+    def validate_hosts(hosts):
+        """Make sure the hosts list is reasonable."""
+        host_parts = hosts.split(',')
```

Comment:
If it is decided to not use relationship with hosts then this will be considering spaces as valid information. I think this validation could be improved. It seems that it will accept values like `ghj,kl;,#$%` it is a comma separated value but the parts are not valid hosts.

Please excuse me if the hosts parts validation is done somewhere else, I need to catch up how Django and Django REST framework is working now a days.

Reply:
I think Noah was planning to do more of the validation in a follow on  PR per his open PR remarks, but we should go ahead and open an issue describing that work in that case.